### PR TITLE
feat(ui5-select): Change focus of options on keypress

### PR DESCRIPTION
**Pull request by IlianaB**
Implement changing of focus of options on keypress - typing letters now allows focusing different options of the ui5-select.

Fixes: #3476

### CODE REVIEW

**Comment by ilhan007** [comment]
Eslint errros, you can run yarn lint locally in the root of the project:

```js
/home/travis/build/SAP/ui5-webcomponents/packages/main/src/Select.js
  480:3  error  'sText' is never reassigned. Use 'const' instead          prefer-const
  494:7  error  'oItemToSelect' is never reassigned. Use 'const' instead  prefer-const
  496:1  error  Expected indentation of 2 tabs but found 3                indent
  497:1  error  Expected indentation of 3 tabs but found 4                indent
  499:1  error  Expected indentation of 3 tabs but found 4                indent
  501:1  error  Expected indentation of 3 tabs but found 4                indent
  502:1  error  Expected indentation of 4 tabs but found 5                indent
  503:1  error  Expected indentation of 3 tabs but found 4                indent
  504:1  error  Expected indentation of 2 tabs but found 3                indent
✖ 9 problems (9 errors, 0 warnings)
  8 errors and 0 warnings potentially fixable with the `--fix` option.
  ```
  
 In addition, you can rebase with the master, as some randomly failing tests have been fixed. Could avoid some build failures, that have nothing to do with this change.

**Comment by IlianaB** [comment]
> Eslint errros, you can run yarn lint locally in the root of the project:
> 
> ```js
> /home/travis/build/SAP/ui5-webcomponents/packages/main/src/Select.js
>   480:3  error  'sText' is never reassigned. Use 'const' instead          prefer-const
>   494:7  error  'oItemToSelect' is never reassigned. Use 'const' instead  prefer-const
>   496:1  error  Expected indentation of 2 tabs but found 3                indent
>   497:1  error  Expected indentation of 3 tabs but found 4                indent
>   499:1  error  Expected indentation of 3 tabs but found 4                indent
>   501:1  error  Expected indentation of 3 tabs but found 4                indent
>   502:1  error  Expected indentation of 4 tabs but found 5                indent
>   503:1  error  Expected indentation of 3 tabs but found 4                indent
>   504:1  error  Expected indentation of 2 tabs but found 3                indent
> ✖ 9 problems (9 errors, 0 warnings)
>   8 errors and 0 warnings potentially fixable with the `--fix` option.
> ```
> 
> In addition, you can rebase with the master, as some randomly failing tests have been fixed. Could avoid some build failures, that have nothing to do with this change.

Done. It seems like have executed the eslint check in the wrong directory.

**Review by PetyaMarkovaBogdanova — DISMISSED** [review]
Looks good to me! Works fine with different scenarios.

**Review by fifoosid — CHANGES_REQUESTED** [review]

File: packages/main/src/Select.js [hunk 1]
```diff
@@ -316,6 +316,8 @@ class Select extends UI5Element {
 		this._selectedIndexBeforeOpen = -1;
 		this._escapePressed = false;
 		this._lastSelectedOption = null;
+		this._sTypedChars = "";
```
[the remark this answers is not in the document]
Comment (IlianaB) [reply packages/main/src/Select.js]: Fixed at all places now.

File: packages/main/src/Select.js [hunk 2]
```diff
@@ -458,11 +460,59 @@ class Select extends UI5Element {
 			this._handleEndKey(event);
 		} else if (isEnter(event)) {
 			this._handleSelectionChange();
+		} else if (event.keyCode >= 65 && event.keyCode <= 90) {
```
[the remark this answers is not in the document]
Comment (IlianaB) [reply packages/main/src/Select.js]: Not specified in the BLI, but improved now, as, yes, it makes a lot more sense! Thanks.

**Review by fifoosid — APPROVED** [review]

**Pull request merged**